creator-node/src/snapbackSM/snapbackSM.test.js

```diff
@@ -0,0 +1,228 @@
+      assert.strictEqual(
+        e.message,
+        '[computeSyncModeForUserAndReplica] Error: Missing or invalid params'
+      )
+    }
+  })
+
+  it('Returns SyncMode.None if all data is equal', async () => {
+    primaryClock = 10
+    secondaryClock = primaryClock
+    primaryFilesHash = '0x123'
+    secondaryFilesHash = primaryFilesHash
+
+    const syncMode = await computeSyncModeForUserAndReplica({
+      wallet,
+      primaryClock,
+      secondaryClock,
+      primaryFilesHash,
+      secondaryFilesHash
+    })
+
+    assert.strictEqual(syncMode, SyncMode.None)
+  })
+
+  it('Returns SyncMode.None if clocks and filesHashes equal', async () => {
+    primaryClock = 10
+    secondaryClock = primaryClock
+    primaryFilesHash = '0x123'
+    secondaryFilesHash = primaryFilesHash
+
+    const syncMode = await computeSyncModeForUserAndReplica({
+      wallet,
+      primaryClock,
+      secondaryClock,
+      primaryFilesHash,
+      secondaryFilesHash
+    })
+
+    assert.strictEqual(syncMode, SyncMode.None)
+  })
+
+  it('Returns SyncMode.PrimaryShouldSync if clocks equal and filesHashes unequal', async () => {
+    primaryClock = 10
+    secondaryClock = primaryClock
+    primaryFilesHash = '0x123'
+    secondaryFilesHash = '0x456'
+
+    const syncMode = await computeSyncModeForUserAndReplica({
+      wallet,
+      primaryClock,
+      secondaryClock,
+      primaryFilesHash,
+      secondaryFilesHash
+    })
+
+    assert.strictEqual(syncMode, SyncMode.PrimaryShouldSync)
+  })
+
+  it('Returns SyncMode.PrimaryShouldSync if clocks equal and filesHashes unequal', async () => {
+    primaryClock = 10
+    secondaryClock = primaryClock
+    primaryFilesHash = '0x123'
+    secondaryFilesHash = '0x456'
+
+    const syncMode = await computeSyncModeForUserAndReplica({
+      wallet,
+      primaryClock,
+      secondaryClock,
+      primaryFilesHash,
+      secondaryFilesHash
+    })
+
+    assert.strictEqual(syncMode, SyncMode.PrimaryShouldSync)
+  })
+
+  it('Returns SyncMode.PrimaryShouldSync if primaryClock < secondaryClock', async () => {
+    primaryClock = 5
+    secondaryClock = 10
+    primaryFilesHash = '0x123'
+    secondaryFilesHash = '0x456'
+
+    const syncMode = await computeSyncModeForUserAndReplica({
+      wallet,
+      primaryClock,
+      secondaryClock,
+      primaryFilesHash,
+      secondaryFilesHash
+    })
+
+    assert.strictEqual(syncMode, SyncMode.PrimaryShouldSync)
+  })
+
+  it('Returns SyncMode.SecondaryShouldSync if primaryClock > secondaryClock & secondaryFilesHash === null', async () => {
+    primaryClock = 10
+    secondaryClock = 5
+    primaryFilesHash = '0x123'
+    secondaryFilesHash = null
+
+    const syncMode = await computeSyncModeForUserAndReplica({
+      wallet,
+      primaryClock,
+      secondaryClock,
+      primaryFilesHash,
+      secondaryFilesHash
+    })
+
+    assert.strictEqual(syncMode, SyncMode.SecondaryShouldSync)
+  })
+
+  describe('primaryClock > secondaryClock', () => {
```

Comment:
if there's a describe block for this event, can we do this for the other events too? e.g. the pc < sc, pc == sc

Reply:
we could but not much point of a describe block with a single test in there